src/ngraph/runtime/executable.hpp

```diff
@@ -80,32 +80,46 @@ class NGRAPH_API ngraph::runtime::Executable
     /// \brief Create an input Tensor
     /// \param input_index The index position in the input Parameter vector. This would be the same
     /// order of Parameters passed into the inputs in the call() method.
+    /// \param memory_pointer A pointer to a buffer used for this tensor. The size of the buffer
+    ///     must be sufficient to contain the tensor. The lifetime of the buffer is the
+    ///     responsibility of the caller.
     /// \returns A Tensor
-    virtual std::shared_ptr<runtime::Tensor> create_input_tensor(size_t input_index);
+    virtual std::shared_ptr<runtime::Tensor> create_input_tensor(size_t input_index,
+                                                                 void* memory_pointer = nullptr);
```

Comment:
If you leave the old API and add these methods (without the "= nullptr" default) as a new API then it will be backwards compatible. As it is these API changes will break the NNP backend, but not in a way that is too difficult to fix, but it is not backwards compatible.